### ISSUE TITLE
Controller is not able to connect to etcd. The current scope has already been entered.

### ISSUE THREAD

**Issue by tobg** [piraeusdatastore/piraeus-operator#113]
Since today linstor is not working anymore. cs controller cannot communicate with etcd.

What does "The current scope has already been entered" means?

kubectl -n piraeus-system exec deployment/piraeus-op-cs-controller -- linstor err show 5FA5BD9C-00000-000001
ERROR REPORT 5FA5BD9C-00000-000001

============================================================

Application:                        LINBIT�� LINSTOR
Module:                             Controller
Version:                            1.9.0
Build ID:                           678acd24a8b9b73a735407cd79ca33a5e95eb2e2
Build time:                         2020-09-23T09:33:23+00:00
Error time:                         2020-11-06 21:18:23
Node:                               piraeus-op-cs-controller-5456d5cddd-nfnfl

============================================================

Reported error:
===============

Category:                           RuntimeException
Class name:                         IllegalStateException
Class canonical name:               java.lang.IllegalStateException
Generated at:                       Method 'checkState', Source file 'Preconditions.java', Line #508

Error message:                      The current scope has already been entered

Call backtrace:

    Method                                   Native Class:Line number
    checkState                               N      com.google.common.base.Preconditions:508
    enter                                    N      com.linbit.linstor.api.LinStorScope:54
    initialize                               N      com.linbit.linstor.systemstarter.PreConnectInitializer:64
    startSystemServices                      N      com.linbit.linstor.core.ApplicationLifecycleManager:87
    start                                    N      com.linbit.linstor.core.Controller:337
    main                                     N      com.linbit.linstor.core.Controller:556

END OF ERROR REPORT.

**Comment by ghernadi** [comment]
> What does "The current scope has already been entered" means?

TL;DR: that's an internal error users should never see :)  
Longer version: Linstor uses scopes for accessing i.e. per-request data like the peer's access context. The same way Linstor also stores in such a scope the current transaction which is used for database-accesses. That means that this error report complains that some other thread/request/something is already in the current scope, most likely already using its own transaction which would conflict in the one the `PreConnectInitializer` tries to use.

As mentioned before this is an error users should never see. The reason for this is, that such error should usually always occur, i.e. also during development where we see and fix this issue...

However, as the stacktrace shows, this is in a very early stage of the controller starting, where not even the REST-API is started. Therefore I highly doubt that there is anything else (like a thread or request) interfering with this. 

In other words, I would be interested in the previous error report `5FA5BD9C-00000-000000` as that could contain the actual error making the shown error report `5FA5BD9C-00000-000001` only a consequential error.

**Comment by tobg** [comment]
Thank you. As I was reinstalled my k8s cluster this issue is no longer an issue.
